src/Solnet.Programs/StakeProgram.cs

```diff
@@ -0,0 +1,365 @@
+            {
+                new_authorized_pubkey = new_authorized_pubkey,
+                stake_authorize = stake_authorize,
+                authority_seed = authority_seed,
+                authority_owner = authority_owner
+            };
+            return new TransactionInstruction
+            {
+                ProgramId = ProgramIdKey.KeyBytes,
+                Keys = keys,
+                Data = StakeProgramData.EncodeAuthorizeWithSeedData(authorizeWithSeed)
+            };
+        }
+        public static TransactionInstruction InitializeChecked(PublicKey stake_pubkey, Authorized authorized)
+        {
+            List<AccountMeta> keys = new()
+            {
+                AccountMeta.Writable(stake_pubkey, false),
+                AccountMeta.ReadOnly(SysVarRentKey, false),
+                AccountMeta.ReadOnly(authorized.staker, false),
+                AccountMeta.ReadOnly(authorized.withdrawer,true)
+            };
+            return new TransactionInstruction
+            {
+                ProgramId = ProgramIdKey.KeyBytes,
+                Keys = keys,
+                Data = StakeProgramData.EncodeInitializeCheckedData()
+            };
+        }
+        public static TransactionInstruction AuthorizeChecked(PublicKey stake_pubkey, PublicKey authorized_pubkey, PublicKey new_authorized_pubkey, StakeAuthorize stake_authorize, PublicKey custodian_pubkey)
+        {
+            List<AccountMeta> keys = new()
+            {
+                AccountMeta.Writable(stake_pubkey, false),
+                AccountMeta.ReadOnly(SysVarClockKey, false),
+                AccountMeta.ReadOnly(authorized_pubkey, true),
+                AccountMeta.ReadOnly(new_authorized_pubkey,true)
+            };
+            if (custodian_pubkey != null)
+            {
+                keys.Add(AccountMeta.ReadOnly(custodian_pubkey, true));
+            }
+            return new TransactionInstruction
+            {
+                ProgramId = ProgramIdKey.KeyBytes,
+                Keys = keys,
+                Data = StakeProgramData.EncodeAuthorizeCheckedData(stake_authorize)
+            };
+        }
+        public static TransactionInstruction AuthorizeCheckedWithSeed(PublicKey stake_pubkey, PublicKey authority_base, string authority_seed, PublicKey authority_owner, PublicKey new_authorized_pubkey, StakeAuthorize stake_authorize, PublicKey custodian_pubkey)
+        {
+            List<AccountMeta> keys = new()
+            {
+                AccountMeta.Writable(stake_pubkey, false),
+                AccountMeta.ReadOnly(authority_base, true),
+                AccountMeta.ReadOnly(SysVarClockKey, false),
+                AccountMeta.ReadOnly(new_authorized_pubkey, true)
+            };
+            if (custodian_pubkey != null)
+            {
+                keys.Add(AccountMeta.ReadOnly(custodian_pubkey, true));
+            }
+            AuthorizeCheckedWithSeedArgs authorizeWithSeed = new AuthorizeCheckedWithSeedArgs
```

Comment:
In general I have not done these ```SomeTypeArgs``` and have just passed the arguments directly to the method that encodes the data for the instruction. I would prefer if we kept this with a concise style all throughout the codebase, not having to do them for every program also makes the codebase leaner despite some of the function signatures being a little more verbose sometimes

Reply:
Eliminated